src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1745,8 +1742,8 @@ static int add_object_vtable_internal(
         assert_return(interface_name_is_valid(interface), -EINVAL);
         assert_return(vtable, -EINVAL);
         assert_return(vtable[0].type == _SD_BUS_VTABLE_START, -EINVAL);
-        assert_return(vtable[0].x.start.element_size == VTABLE_ELEMENT_SIZE_ORIGINAL ||
-                      vtable[0].x.start.element_size == VTABLE_ELEMENT_SIZE,
+        assert_return(vtable[0].x.start.element_size == VTABLE_ELEMENT_SIZE_221 ||
+                      vtable[0].x.start.element_size >= VTABLE_ELEMENT_SIZE_242,
                       -EINVAL);
```

Comment:
hmm, but now we'd silently accept vtables that might use future features we don't support which might create problems.

I figure there should be a vtable_features() check here, that validates that there are no bits set in it we don't understand

Reply:
No, that would be useless. If the table is extended in a backwards-compatible way, the old code will work fine with the new tables. If the table is extended in a non-backwards compatible way, we'll need to add a new symbol and/or bump symbol versions.

Consider the last addition: it didn't work in this way because of the overzealous size check in the old code. But apart from that, it should have been fully backwards-compatible. If we were later to e.g. add another field in the structure that would provide some additional information (but the table would still be usable without it), the code in this PR should work fine with it.